From cf020f133487d36a8b1d9cfd16ec456f7f07952e Mon Sep 17 00:00:00 2001
From: Marcus Boerger <helly@php.net>
Date: Sat, 19 Mar 2005 15:06:39 +0000
Subject: [PATCH] - Fix #31185

---
 Zend/zend_object_handlers.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/Zend/zend_object_handlers.c b/Zend/zend_object_handlers.c
index ca2de17f2fcb..43f9d472e858 100644
--- a/Zend/zend_object_handlers.c
+++ b/Zend/zend_object_handlers.c
@@ -476,13 +476,11 @@ static void zend_std_unset_property(zval *object, zval *member TSRMLS_DC)
 static void zend_std_unset_dimension(zval *object, zval *offset TSRMLS_DC)
 {
 	zend_class_entry *ce = Z_OBJCE_P(object);
-	zval *retval;
 	
 	if (instanceof_function_ex(ce, zend_ce_arrayaccess, 1 TSRMLS_CC)) {
 		SEPARATE_ARG_IF_REF(offset);
-		zend_call_method_with_1_params(&object, ce, NULL, "offsetunset", &retval, offset);
+		zend_call_method_with_1_params(&object, ce, NULL, "offsetunset", NULL, offset);
 		zval_ptr_dtor(&offset);
-		zval_ptr_dtor(&retval);
 	} else {
 		zend_error(E_ERROR, "Cannot use object of type %s as array", ce->name);
 	}
