From 7412202c43e74d24c55669dedc36542e43b4f2bb Mon Sep 17 00:00:00 2001
From: Antony Dovgal <tony2001@php.net>
Date: Thu, 11 May 2006 22:41:37 +0000
Subject: [PATCH] no need to destroy the zval here

---
 Zend/zend.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Zend/zend.c b/Zend/zend.c
index b1a94ce2abeb..cf54baaa583b 100644
--- a/Zend/zend.c
+++ b/Zend/zend.c
@@ -248,7 +248,6 @@ ZEND_API void zend_make_printable_zval(zval *expr, zval *expr_copy, int *use_cop
 					}
 				}
 				zend_error(EG(exception) ? E_ERROR : E_RECOVERABLE_ERROR, "Object of class %s could not be converted to string", Z_OBJCE_P(expr)->name);
-				zval_dtor(expr_copy);
 				expr_copy->value.str.len = 0;
 				expr_copy->value.str.val = STR_EMPTY_ALLOC();
 			}
