From d09cdd9f71f34deab4b99f4e63523fb94164a724 Mon Sep 17 00:00:00 2001
From: Dmitry Stogov <dmitry@php.net>
Date: Wed, 8 Jun 2005 18:10:28 +0000
Subject: [PATCH] Fixed valgrind errors

---
 Zend/zend_compile.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Zend/zend_compile.c b/Zend/zend_compile.c
index 5d0eb4bcf9d4..4cd638513872 100644
--- a/Zend/zend_compile.c
+++ b/Zend/zend_compile.c
@@ -2582,7 +2582,8 @@ void zend_do_end_class_declaration(znode *class_token, znode *parent_token TSRML
 
 	/* Inherit interfaces */
 	if (ce->num_interfaces > 0) {
-		ce->interfaces = (zend_class_entry **) erealloc(ce->interfaces, sizeof(zend_class_entry *)*ce->num_interfaces);
+		ce->interfaces = (zend_class_entry **) emalloc(sizeof(zend_class_entry *)*ce->num_interfaces);
+		memset(ce->interfaces, 0, sizeof(zend_class_entry *)*ce->num_interfaces);
 	}
 	if (!(ce->ce_flags & (ZEND_ACC_INTERFACE|ZEND_ACC_EXPLICIT_ABSTRACT_CLASS))
 		&& ((parent_token->op_type != IS_UNUSED) || (ce->num_interfaces > 0))) {
